1110 simulation sync status after project conversion

rwmcintosh opened the pull request:
Fixes #1110 

# Description
Changes in V11 simulations cannot be traced back to the building blocks when converted to V12. This PR shows a notification at the time of project conversion and another notification if the simulation changes are committed to building blocks.

## Type of change

Please mark relevant options with an `x` in the brackets.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires documentation changes (link at least one [user](https://github.com/Open-Systems-Pharmacology/docs) or [developer](https://github.com/Open-Systems-Pharmacology/developer-docs) documentation issue):
- [ ] Algorithm update - updates algorithm documentation/questions/answers etc.
- [ ] Other (please describe):

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Integration tests
- [x] Unit tests
- [x] Manual tests 
- [ ] No tests required

# Reviewer checklist

Mark everything that needs to be checked before merging the PR.

- [ ] Check if the code is well documented
- [ ] Check if the behavior is what is expected
- [ ] Check if the code is well tested
- [ ] Check if the code is readable and well formatted
- [ ] Additional checks (document below if any)
- [ ] Check if documentation update issue(s) are created if the option `This change requires a documentation update` above is selected

# Screenshots (if appropriate):
![image](https://github.com/user-attachments/assets/e46258c3-fb2c-4000-b0ad-01fa2a1eda04)

![image](https://github.com/user-attachments/assets/9e2d49f3-3bd3-42dc-81e6-ee22c448ddb3)

# Questions (if appropriate):

msevestre commented on src/MoBi.Core/Domain/Model/MoBiSimulation.cs:
```diff
@@ -49,6 +49,7 @@ public interface IMoBiSimulation : IWithDiagramFor<IMoBiSimulation>, ISimulation
       IReadOnlyList<IBuildingBlock> BuildingBlocks();
 
       IReadOnlyCollection<OriginalQuantityValue> OriginalQuantityValues { get; }
+      bool HasUntraceableChanges { get; set; }
```
I think this guy deserves some comments.
Also it sucks to have a property here that is just required for a one time project conversion...

msevestre commented on src/MoBi.Core/Commands/UpdateSimulationCommand.cs:
```diff
@@ -22,28 +22,32 @@ public class UpdateSimulationCommand : MoBiReversibleCommand
       {
          _hasChanged = wasChanged;
+         _hasUntraceableChanges = hasUntraceableChanges;
+         _hadUntraceableChanges = simulationToUpdate.HasUntraceableChanges;
```
it took me  a minute to see the `has` vs `had`

msevestre commented on src/MoBi.Core/Serialization/Converter/v12/Converter11To12.cs:
```diff
@@ -28,10 +30,34 @@ public Converter11To12(Converter110To120 coreConverter)
          {
+            var simulationChanges = simulationNode.DescendantsAndSelfNamed("BuildConfiguration").Any(hasSimulationChanges);
+            if (simulationChanges)
+               simulationNode.AddAttribute("hasUntraceableChanges", true.ToString());
```
true to string? lol. what about "true"?

msevestre replied:
yeah... I don't have a better idea but this feels kind of wrong (The property not the boolean)

msevestre commented on src/MoBi.Presentation/Presenter/Main/NotificationPresenter.cs:
```diff
@@ -250,11 +249,25 @@ public void ShowContextMenu(IViewItem notificationItem, Point popupLocation)
+      public void Handle(ShowProjectConversionNotificationsEvent eventToHandle)
+      {
+         Handle(eventToHandle as ShowNotificationsEvent);
+         notifyForUntraceableChanges(eventToHandle.NotificationMessages.Select(x => x.Object as MoBiSimulation).Where(x => x != null && x.HasUntraceableChanges).Distinct().ToList());
```
what about we create a local variables here?

msevestre replied:
cannot we have an objec that has the simulation and this flag just for the conversion? instead of havign it into the simulation

rwmcintosh replied:
The simulation is keeping this flag because the sync status remains red in the explorer until it is reconfigured. That's also why you get the reminder each time you commit and not only once during conversion. I'm not sure if that's necessary but it seems right. The simulation is not in sync and it will never be without reconfigure.

msevestre commented on src/MoBi.Presentation/Tasks/SimulationCommitTask.cs:
```diff
@@ -94,9 +95,18 @@ public IMoBiCommand CommitSimulationChanges(IMoBiSimulation simulationWithChange
          _context.PublishEvent(new SimulationStatusChangedEvent(simulationWithChanges));
+
+         if (simulationWithChanges.HasUntraceableChanges)
+            showWarningDialogForUntraceableChanges(simulationWithChanges);
```
why are we showing this again here? Isn't the conversion warning enough?

rwmcintosh commented:
Updated warning message during conversion
![image](https://github.com/user-attachments/assets/8e0f4f76-2e25-4420-ac6a-252671112690)

Updated commit message as an error and modified text. Also, no commits to BB's are done when this simulation has this status.
![image](https://github.com/user-attachments/assets/eacce16d-9922-43d1-ad1a-34d9e557fd25)
